AEPIdentity/Sources/Identity.swift

```diff
@@ -55,41 +55,50 @@ import Foundation
         } else if event.type == EventType.configuration, event.source == EventSource.requestIdentity {
-            return MobileIdentities().areSharedStatesReady(event: event, sharedStateProvider: getSharedState(extensionName:event:))
+            let areSharedStateReady = MobileIdentities().areSharedStatesReady(event: event, sharedStateProvider: getSharedState(extensionName:event:))
+
+            if !areSharedStateReady {
+                Log.trace(label: "\(name):\(#function)", "Waiting for the Mobile Identities states to be set before processing [event:(\(event.name)) id:(\(event.id)].")
+            }
+
+            return areSharedStateReady
         } else if event.type == EventType.identity, event.source == EventSource.requestIdentity, ( event.baseUrl != nil ||  event.urlVariables ) {
 
             // analytics shared state will be null if analytics extension is not registered. Wait for analytics shared only if the status is pending or none
             if let analyticsSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.ANALYTICS, event: event), analyticsSharedState.status != .set {
-                Log.trace(label: "\(name):\(#function)", "Waiting for the Analytics shared state to be set.")
+                Log.trace(label: "\(name):\(#function)", "Waiting for the Analytics shared state to be set before processing [event:(\(event.name)) id:(\(event.id)].")
                 return false
             }
         }
 
-        return getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: event)?.status == .set
-    }
+        let isConfigSharedStateSet = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: event)?.status == .set
 
-    /// Determines if Identity is ready to handle events, this is determined by if the Identity extension has booted up
-    /// - Parameter event: An `Event`
-    /// - Returns: True if we can process events, false otherwise
-    private func canProcessEvents(event: Event) -> Bool {
-        guard let state = state else { return false }
-        guard !state.hasBooted else { return true } // we have booted, return true
-
-        // fetch latest configuration when booting up
-        guard let configSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value else { return false }
-        // attempt to bootup
-        if state.bootupIfReady(configSharedState: configSharedState, event: event) {
-            createSharedState(data: state.identityProperties.toEventData(), event: nil)
+        if !isConfigSharedStateSet {
+            Log.trace(label: "\(name):\(#function)", "Waiting for the Configuration shared state to be set before processing [event:(\(event.name)) id:(\(event.id)].")
         }
 
-        return false // cannot handle any events until we have booted
-
+        return isConfigSharedStateSet
```

Comment:
What other events are handled by this case? Does resetIdentities for example also wait for config?

Reply:
@emdobrin we wait for config for almost all events at present, except for have isSyncEvent or genericIdentity event type where we look for readyForSyncIdentifiers check

Reply:
Ok, we may need to evaluate the other events also, e.g. resetIdentities should not have a dependency on config, but I think we can keep that outside of this PR.